lib/modules/manager/cargo/artifacts.ts

```diff
@@ -120,14 +121,23 @@ async function updateArtifactsImpl(
     if (isLockFileMaintenance) {
       await cargoUpdate(packageFileName, true, config.constraints?.rust);
     } else {
-      const missingDep = updatedDeps.find((dep) => !dep.lockedVersion);
-      if (missingDep) {
-        // If there is a dependency without a locked version then log a warning
-        // and perform a regular workspace lockfile update.
-        logger.warn(
-          { dependency: missingDep.depName },
-          'Missing locked version for dependency',
-        );
+      const nonCrateDep = updatedDeps.find(
```

Comment:
```suggestion
      const nonCrateDep = updatedDeps.some(
```
Since, we only need to value not the object